eng/pipelines/templates/steps/vcpkg.yml

```diff
@@ -17,45 +11,27 @@ steps:
       Write-Host "##vso[task.prependpath]$TargetPath"
       Write-Host "##vso[task.setvariable variable=VCPKG_INSTALLATION_ROOT]$TargetPath"
       Write-Host "##vso[task.setvariable variable=VcpkgCommit]$VcpkgCommit"
+      Write-Host "##vso[task.setvariable variable=VCPKG_BINARY_SOURCES]clear;x-azblob,https://cppvcpkgcache.blob.core.windows.net/public-vcpkg-container,,read"
     displayName: Set Vcpkg Variables
     condition: >-
-      and(
-        succeeded(),
-        not(eq(variables['${{ parameters.DependenciesVariableName }}'], ''))
-      )
+      succeeded()
 
-  # This task times out after 10 minutes. An analysis of a subset of Vcpkg Cache
-  # runs showed that after 10 minutes the cache step is very unlikely to
-  # succeed.
-  - task: Cache@2
-    inputs:
-      key: >-
-        $(Agent.JobName)
-        | "$(VcpkgCommit)"
-        | $(Agent.Os)
-      path: $(VCPKG_INSTALLATION_ROOT)
-      cacheHitVar: VcpkgRestoredFromCache
-    displayName: Vcpkg Cache
-    timeoutInMinutes: 10
+  - pwsh: |
+      Write-Host "Setting vcpkg binary cache to read and write"
+      Write-Host "##vso[task.setvariable variable=VCPKG_BINARY_SOURCES;issecret=true]clear;x-azblob,https://cppvcpkgcache.blob.core.windows.net/public-vcpkg-container,$env:VCPKG_CACHE_SAS,readwrite"
```

Comment:
You should probably at least check that VCPKG_CACHE_SAS has a value and if not give an error message as that might help diagnostics if something is not setup correctly.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/3418